### ISSUE TITLE
[explicit-module-boundary-types] Not usable, because I sometimes need a parameter to be explicitly `any`

### ISSUE THREAD

**Issue by Shinigami92** [typescript-eslint/typescript-eslint#2064]
**Repro**

```JSONC
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": ["./tsconfig.json"]
  },
  "plugins": ["@typescript-eslint"],
  "rules": {
    // Default `explicit-module-boundary-types` from recommended
  }
}
```

```TS
export class MyLogger {
  public log(message?: any, ...optionalParams: any[]): void {
    this.message('log', message, ...optionalParams);
  }

  private message(level: 'log', message?: any, ...optionalParams: any[]): void {
    console[level](message, ...optionalParams);
  }
}
```

Real world example: [prettier/plugin-pug:src/logger.ts#L33-L51](https://github.com/prettier/plugin-pug/blob/69f4908fe66e39100245ac30263831751fae7ad9/src/logger.ts#L33-L51)

**Expected Result**

No warning, because I explicitly added the typedef `any` to parameter `message`

**Actual Result**

<img src="https://user-images.githubusercontent.com/7195563/82647504-94ed1380-9c16-11ea-956f-046902d12048.png" width="500px">

**Additional Info**

Currently it seems that I have to disable this rule in my repo as there is no option to allow `any` as a _valid_ type

[prettier/plugin-pug:.eslintrc.json#L32](https://github.com/prettier/plugin-pug/blob/69f4908fe66e39100245ac30263831751fae7ad9/.eslintrc.json#L32)

If there is an option that I missed, please improve the documentation: [(1f3c344) docs/rules/explicit-module-boundary-types.md](https://github.com/typescript-eslint/typescript-eslint/blob/1f3c34426cea6d04df2393032e0728ade7390d3c/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.0` |
| `@typescript-eslint/parser`        | `3.0.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.0.0` |
| `node`                             | `12.16.3` |
| `npm`                              | `6.14.4` |

**Comment by TrejGun** [comment]
same here
for example custom validation function should accept anything
here is a good example
https://github.com/typestack/class-validator/blob/master/sample/sample6-custom-decorator/IsLongerThan.ts#L22

**Comment by Raynos** [comment]
You can replace `any` with `unknown`. This should fix the linter warning.

Using explicit `any` is not recommened, explicit `unknown` is better.

**Comment by Shinigami92** [comment]
@Raynos This has indeed fixed my problem

Should we improve the documentation of the rule and give other consumers a hint?

**Comment by TrejGun** [comment]
```
export class Test1 {
  public async test(value: any): Promise<boolean> {
    return Promise.resolve(value as boolean);
  }
}

class Test2 {
  public async test(value: any): Promise<boolean> {
    return Promise.resolve(value as boolean);
  }
}
```

in this example Test1.test will be reported but Test2.test will not. I guess it is bacause of export keyword. Is it correct behavior?

**Comment by bradzacher** [comment]
@TrejGun yes.
In the past many, many people found that `explicit-function-return-type` was too restrictive and didn't like that it was recommended.

A lot of people don't like doing this explicit definition all of the time.

We still think providing explicit types is a best practice, so we have instead switched to `explicit-module-boundary-types`, which enforces that your module boundaries are well typed.

Within a module's internals you can do whatever you like, but at the module boundary only will it enforce your parameters are clearly and explicitly typed.

---

`unknown` is the safe variant to use. You should not use `any` because it's completely unsafe.

If you're using the recommended set, it should also have `no-explicit-any`, so you should be warned against using `any` anyway.

I've updated the docs to add an example suggesting using `unknown`
